x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/AuthenticationService.java

```diff
@@ -488,7 +492,13 @@ private void consumeToken(AuthenticationToken token) {
                 final IteratingActionListener<User, Realm> authenticatingListener =
                     new IteratingActionListener<>(ContextPreservingActionListener.wrapPreservingContext(ActionListener.wrap(
                         (user) -> consumeUser(user, messages),
-                        (e) -> listener.onFailure(request.exceptionProcessingRequest(e, token))), threadContext),
+                        (e) -> {
+                            if (e != null) {
+                                listener.onFailure(request.exceptionProcessingRequest(e, token));
+                            } else {
+                                listener.onFailure(request.authenticationFailed(authenticationToken));
```

Comment:
This is more of a question: why were `token` and `authenticationToken` both used in differently places? Should we prefer one over the other?

Reply:
:+1: I think we should be consistent here.

Reply:
This now uses `token` in both instances.
It's probably more idiomatic like this. If you feel otherwise let me know.